### PR TITLE
Feat/discussion close with disclaimer

### PR DESCRIPTION
**Pull request by ksowndar**
Closed field is added in Post model
Patch is created and close method is added in controller

### CODE REVIEW

File: app/controllers/comments_controller.rb [hunk 1]
```diff
@@ -1,23 +1,17 @@
...
-          render "comments/create"
-        else
-          # unable to save
-        end
-      }
-    end
+    if @comment.save
+      redirect_to community_post_path(@comment.post_id, @comment.post), flash: {success: "comment was created successfully!" }
+    else
+      redirect_to community_post_path(@comment.post_id, @comment.post), flash: { danger: "comment was not saved!" }
```
Comment (iMagesh) [inline app/controllers/comments_controller.rb:9]: this seems to have a duplicate of the redirect_to, can you make it DRY?

File: app/javascript/packs/application.js [hunk 2]
```diff
@@ -14,6 +14,7 @@ require("../includes/vote.js")
 require("../includes/backtotop.js")
 require("../includes/infinitescroll.js")
 require("../includes/navbar.js")
+require("../includes/closediscussion.js")
```
Comment (iMagesh) [inline app/javascript/packs/application.js:17]: File name does not follow our conventions. Two words cannot be used like this, remember?

File: app/views/posts/_comments.html.erb [hunk 3]
```diff
@@ -1,6 +1,34 @@
...
+<% end %> 
+
+<script>
+  document.querySelectorAll('.comment-form-display').forEach((el) => {
+    el.addEventListener('click', (ev) => {
+     ev.preventDefault();
+     el.nextElementSibling.style = 'display: block;'
+    })
+  })
+</script>
```
Comment (iMagesh) [inline app/views/posts/_comments.html.erb:34]: Please move scripts to a separate js file. Don't include js code like this in html.erb.

File: app/views/posts/_list.html.erb [hunk 4]
```diff
@@ -24,3 +23,9 @@
 <% end %>
 <button onclick="topFunction()" id="myBtn" title="Go to top">Top</button>
 
+
+
+
+
+
+
```
Comment (iMagesh) [inline app/views/posts/_list.html.erb:31]: why do you have a lot of empty space here?

File: app/views/posts/show.html.erb [hunk 5]
```diff
@@ -1,3 +1,17 @@
...
+   $('#close-link').click(function(event){
+     console.log("hi");
+     $("#disclaimer").show();
+     $("#comment-hide").hide();
+     $("#close-link").hide();
+     alert('The discussion is closed.No one able to comment on your discussion');
+    event.preventDefault(); 
+ });
+});
+</script>
```
Comment (iMagesh) [inline app/views/posts/show.html.erb:14]: Please remove js script from html files

File: app/javascript/packs/application.js (hunk 2 above)
Comment (ksowndar) [reply app/javascript/packs/application.js:17]: Ok Magesh

**Review by iMagesh — CHANGES_REQUESTED** [review]
I have added comments, please look into it and make the changes.

File: app/controllers/comments_controller.rb (hunk 1 above)
Comment (ksowndar) [reply app/controllers/comments_controller.rb:9]: Yes Magesh

File: app/views/posts/_comments.html.erb (hunk 3 above)
Comment (ksowndar) [reply app/views/posts/_comments.html.erb]: js files Moved to separate file

File: app/views/posts/_list.html.erb (hunk 4 above)
Comment (ksowndar) [reply app/views/posts/_list.html.erb]: Deleted those empty lines

File: app/views/posts/show.html.erb (hunk 5 above)
Comment (ksowndar) [reply app/views/posts/show.html.erb]: It is added into seperate files

**Comment by ksowndar** [comment]
Resolved all the comments. Now it's ready to Merge

File: app/javascript/includes/form.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+document.querySelectorAll('.comment-form-display').forEach((el) => {
```
Comment (iMagesh) [inline app/javascript/includes/form.js:1]: file name seems to be very generic, can you change it to something better?

**Comment by iMagesh** [comment]
I have added one more comment, please fix it and then we can merge

**Pull request merged**